pkg/controller/upload-controller.go

```diff
@@ -515,7 +516,12 @@ func (r *UploadReconciler) createUploadPod(args UploadPodArgs) (*v1.Pod, error)
 		return nil, err
 	}
 
-	pod := r.makeUploadPodSpec(args, podResourceRequirements)
+	workloadNodePlacement, err := GetWorkloadNodePlacement(r.client)
+	if err != nil {
+		return nil, err
+	}
+
+	pod := r.makeUploadPodSpec(args, podResourceRequirements, *workloadNodePlacement)
```

Comment:
So we (CDI team) are wildly inconsistent with our makeXXXpod functions. I have seen a MakeCloneSourcePodSpec which exports that function for no apparent reason. I have seen makeImporterPodSpec, and now here r.makeUploadPodSpec. I think they should all be part of their respective reconciler struct. Because that will allow you to make
```go
workloadNodePlacement, err := GetWorkloadNodePlacement(r.client)
if err != nil {
  return nil, err
}
```
part of the r.makeXXXPodSpec function. Right now you are looking up the placement outside of that function, for the sole purpose of passing it to the function. Its not used anywhere else, so it makes sense if that code is part of the r.makeXXXPodSpec function.

So I request that you make those functions consistent, and put the code inside the functions.